src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,48 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
     private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
 
     public void add(E element) {
+        Lock lock = readWriteLock.writeLock();
+        lock.lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.unlock();
+        }
         // write your code here
     }
 
     public E get(int index) {
+        Lock lock = readWriteLock.readLock();
+        lock.lock();
+        E result;
+        try {
+            result = list.get(index);
+        } finally {
+            lock.unlock();
+        }
         // write your code here
-        return null;
+        return result;
     }
 
     public int size() {
+        Lock lock = readWriteLock.readLock();
+        lock.lock();
+        int result;
+        try {
+            result = list.size();
+        } finally {
+            lock.unlock();
+        }
```

Comment:
```suggestion
           return list.size();
        } finally {
            lock.unlock();
        }
```

Reply:
but why if we first
is executed finally before the return
it turns out that we are extracting data from an already unlocked thread